### DISCUSSION TITLE
Use ENTER to exit filter input mode?

### DISCUSSION THREAD

**Discussion by vlappa** [jarun/nnn#1849] [General]
When you filter in nnn with '/', hitting ENTER does a action on the selected file. 

You can't use the vim keys to scroll down the filtered list, because then the 'j' or 'k' will be added to the filter input. You can only scroll down with the up/down arrow keys.

A alternative would be to use ENTER to set the filter input first. So a ENTER doesn't do a action on the selection, but exits the filter input mode. The filtered list is showed in nnn. One can then use the vim keys to scroll down the filtered list.

Ok, when you scroll down one step first, it exits the filter input and you can use the vim keys again.

I think I would also use ENTER to exit the filter input mode.

**Comment by N-R-K** [comment]
> I think I would also use ENTER to exit the filter input mode.

Then you'd have to press ENTER twice just to select a directory to enter. Such a change would pretty much make type to nav unusable for me.

And besides, exiting something with enter is weird. You generally use ESC to exit and you can use it to exit filter mode as well.

**Comment by vlappa** [comment]
Then exiting might not be the good word. :) 

Set the word to filter. 

I constantly want to use the vim keys to scroll when filtering, but then it changes the filter input. 

ESC is far away to end something like filter input IMO. Same for arrow keys.

**Reply by N-R-K** [reply]
> ESC is far away

Remapping CapsLock to Escape is a pretty common thing to do for people with keyboard based workflow.

There's also `ctrl+]` which sends the same code as `esc` in terminals.

**Comment by vlappa** [comment]
Alt+Esc is ending filter mode (unfilter)? This does quit nnn when the filter word is not 'set' (you first have to hit Esc or Arrow keys). I think there is room for improvements here. 

Maybe a troublesome key for some terminals (?), but I would use backspace probably to unfilter.
